server/controllers/notifications.js

```diff
@@ -0,0 +1,77 @@
+const Notification = require("../models/Notification");
+const asyncHandler = require("express-async-handler");
+
+//get all notifications
+exports.getAllNotifications = asyncHandler(async (req, res, next) => {
+let notifications = await Notification.find()
+
+if(!notifications){
+    res.status(404)
+    throw new Error('No notifications found!')
+}
+// can we shorten it this way since the key and value are the same?
+res.status(200).json({notifications})
+});
+
+exports.getAllUnreadNotifications = asyncHandler(async (req, res, next) => {
+    // feel like this is wrong
+let unreadNotifications = await Notification.find({read: false})
+if(!unreadNotifications){
+    res.status(404)
+    throw new Error('no unread notifications')
+}
+res.status(200).json({unreadNotifications})
+});
+
+// creating notification
+exports.createNewNotification = asyncHandler(async (req, res, next) => {
```

Comment:
I'm not sure this makes sense as a route, may have been described that way in the task though. Maybe makes more sense to make this a method on the Notification model, that way if we want to add more functionality (for example socket integration) to the notification it's already consolidated in one place.